main.go

```diff
@@ -3,74 +3,71 @@ package main
 import (
 	"bytes"
 	"context"
-	"fmt"
+	"flag"
+	"github.com/Riskified/worker_runner/internal/healthcheck"
+	log "github.com/sirupsen/logrus"
 	"io"
-	"log"
-	"net/http"
 	"os"
 	"os/exec"
 	"sigs.k8s.io/controller-runtime/pkg/manager/signals"
-	"sync"
 	"syscall"
 )
 
-func healthcheck(w http.ResponseWriter, req *http.Request) {
-	log.Print("got healthcheck")
-	fmt.Fprintf(w, "ok\n")
-}
 
-func startWorker(stopChan <-chan struct{}, wg *sync.WaitGroup, command string, args ...string) {
+
+func startWorker(stopChan <-chan struct{}, command string, args ...string) (doneChan <-chan struct{}) {
 	cmd := exec.Command(command, args...)
 
 	var stdoutBuf, stderrBuf bytes.Buffer
 	cmd.Stdout = io.MultiWriter(os.Stdout, &stdoutBuf)
 	cmd.Stderr = io.MultiWriter(os.Stderr, &stderrBuf)
 
+	done := make(chan struct{})
+
 	go func() {
-		err := cmd.Run()
+		err := cmd.Start()
 		if err != nil {
-			log.Fatalf("cmd.Run() failed with %s\n", err)
+			log.WithField("error", err).Error("Failed starting command")
+			close(done)
```

Comment:
If you put `defer close(done)` at the top of the go routine (I'm not sure you're going to need that)